pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -2697,8 +2697,27 @@ self: super: {
   # https://github.com/kseo/sfnt2woff/issues/1
   sfnt2woff = doJailbreak super.sfnt2woff;
 
-  # 2023-03-05: restrictive bounds on base https://github.com/diagrams/diagrams-gtk/issues/11
-  diagrams-gtk = doJailbreak super.diagrams-gtk;
+  # 2023-06-28: https://github.com/diagrams/diagrams-builder/pull/36
+  diagrams-builder =
+    overrideCabal (drv: assert drv.version == "0.8.0.5" && drv.revision == "5"; {
+      patches = [ ./patches/diagrams-builder-fix-ps.patch ];
```

Comment:
We generally try not to carry around patches in Nixpkgs.  Is there some way you can pull this from an upstream PR?

Reply:
Sadly, I don't think so. The upstream PR is created against the master branch, but the Hackage revision is not on master yet. Moreover, the [hackage revision](https://hackage.haskell.org/package/diagrams-builder-0.8.0.5/revision/5.cabal) contains some changes from the [latest commit](https://github.com/diagrams/diagrams-builder/commit/9ebe6a678b37df5929efc2dca49cda73512b2ccd) (`lens == 5.2`), but not others (`base == 4.17`). Therefore I had to pull the patch from the PR and edit it to account for this partiality myself, otherwise the patch wouldn't apply cleanly. If there is any other way to do this cleaner, I would like to hear for sure!

Reply:
I generally dislike carrying patches for bug fixes (beyond more or less trivial compilation failures) that don't affect nixpkgs itself. Can we just disable the `ps` flag instead?